server/controllers/conversation.js

```diff
@@ -0,0 +1,129 @@
+
+
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/conversation");
+const Message = require("../models/message");
+const Profile = require("../models/Profile");
+// @route Post /create
+// @desc create conversation
+// @access Public
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const userId = await User.findById(req.user.id);
+  if (!userId) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+  const senderId = req.user.id;
+  const { recipientId, text, conversationId } = req.body;
+  if (conversationId) {
+    const message = await Message.create({ senderId, text, conversationId });
+    return res.json({ message });
```

Comment:
Wrap this part in a try/catch so that you return a `500` if there's an error. Please do the same for all other occurences.

Reply:
included the check condition